ckanext/harvest/model/__init__.py

```diff
@@ -79,6 +79,19 @@ def setup():
         if "harvest_job_id_idx" not in index_names:
             log.debug('Creating index for harvest_object')
             Index("harvest_job_id_idx", harvest_object_table.c.harvest_job_id).create()
+        
+        if "harvest_source_id_idx" not in index_names:
+            log.debug('Creating index for harvest source')
+            Index("harvest_source_id_idx", harvest_object_table.c.harvest_source_id).create()
+        
+        if "package_id_idx" not in index_names:
+            log.debug('Creating index for package')
+            Index("package_id_idx", harvest_object_table.c.package_id).create()
+        
+        index_names = [index['name'] for index in inspector.get_indexes("harvest_object_extra")]
+        if "harvest_object_id_idx" not in index_names:
+            log.debug('Creating index for harvest_object_extra')
+            Index("harvest_object_id_idx", harvest_object_extra_table.c.harvest_object_id).create()
```

Comment:
So these indexes will be created at application run time?

Have you done any analysis on how this will affect catalog-classic? How long will it take to build these indexes?

Reply:
@adborden I will finish a document analysis for this to start and issue at upstream. This is not just for GSA as you mention.
This indexes will be created in the application run time **only** if they don't exist ([as the previous indexes](https://github.com/ckan/ckanext-harvest/blob/master/ckanext/harvest/model/__init__.py#L70:L70)). 
I did it in the sandbox and didn't take a lot of time.

Reply:
@adborden here I created a simple document with info to share as a new issue at upstream repo: https://hackmd.io/jYvieE0zQRqOBSAClWj77g

Reply:
A PR was sent to upsteam: https://github.com/ckan/ckanext-harvest/pull/414
I also connected this with an old issue regarding the same problem.